src/neuromaps_prime/transforms/surface.py

```diff
@@ -53,3 +62,438 @@ def surface_sphere_project_unproject(
+    input_file : Path
+        Path to the input GIFTI file (metric or label).
+    source_space : str
+        The source space name.
+    target_space : str
+        The target space name.
+    hemisphere : str
+        The hemisphere ('left' or 'right').
+    output_file_path : str
+        Path to the output GIFTI file.
+    source_density : str, optional
+        Density of the source surface.
+        If None, it will be estimated from the input file.
+    target_density : str, optional
+        Density of the target surface.
+        If None, the highest available density will be used.
+    add_edge : bool, optional
+        Whether to add the resulting transform as an edge in the graph.
+        Default is True.
+
+    Returns:
+    -------
+    result : workbench.MetricResampleOutputs | workbench.LabelResampleOutputs | None
+        The resulting resampled metric or label output.
+
+    Raises:
+    ------
+    ValueError
+        If transformer_type is invalid.
+        If the input file is not found.
+        If the source surface is not found.
+    FileNotFoundError
+        If any input file does not exist.
+        If the output file cannot be created.
+
+    """
+    if transformer_type not in ["metric", "label"]:
+        raise ValueError(
+            f"Invalid transformer_type: {transformer_type}. "
+            "Must be 'metric' or 'label'."
+        )
+    if source_density is None:
+        source_density = estimate_surface_density(input_file)
+
+    transform = _surface_to_surface(
+        graph=graph,
+        source=source_space,
+        target=target_space,
+        density=source_density,
+        hemisphere=hemisphere,
+        output_file_path=output_file_path,
+        add_edge=add_edge,
+    )
+
+    if transform is None:
+        return None
+
+    if target_density is None:
+        target_density = find_highest_density(graph=graph, space=target_space)
+
+    new_sphere_atlas = graph.fetch_surface_atlas(
+        space=target_space,
+        hemisphere=hemisphere,
+        density=target_density,
+        resource_type="sphere",
+    )
+    if new_sphere_atlas is None:
+        raise ValueError(f"No surface atlas found for {target_space}")
+    new_sphere = new_sphere_atlas.fetch()
+
+    current_area_atlas = graph.fetch_surface_atlas(
+        space=source_space,
+        hemisphere=hemisphere,
+        density=source_density,
+        resource_type="midthickness",
+    )
+    if current_area_atlas is None:
+        raise ValueError(f"No midthickness surface found for {source_space}")
+    current_area: Path = current_area_atlas.fetch()
+
+    new_area_atlas = graph.fetch_surface_atlas(
+        space=target_space,
+        hemisphere=hemisphere,
+        density=target_density,
+        resource_type="midthickness",
+    )
+    if new_area_atlas is None:
+        raise ValueError(f"No midthickness surface found for {target_space}")
+    new_area: Path = new_area_atlas.fetch()
```

Comment:
The resource types used here should be a little more flexible - while we generally want to use the midthickness, other surfaces could be used in theory